python/ray/autoscaler/_private/command_runner.py

```diff
@@ -701,7 +701,57 @@ def _docker_expand_user(self, string, any_char=False):
 
         return string
 
-    def run_init(self, *, as_head, file_mounts):
+    def _check_if_container_restart_is_needed(
+            self, image: str, cleaned_bind_mounts: Dict[str, str]) -> bool:
+        re_init_required = False
+        running_image = self.run(
+            check_docker_image(self.container_name),
+            with_output=True,
+            run_env="host").decode("utf-8").strip()
+        if running_image != image:
+            cli_logger.error(
+                "A container with name {} is running image {} instead " +
+                "of {} (which was provided in the YAML)", self.container_name,
+                running_image, image)
+        mounts = self.run(
+            check_bind_mounts_cmd(self.container_name),
+            with_output=True,
+            run_env="host").decode("utf-8").strip()
+        try:
+            active_mounts = json.loads(mounts)
+            active_remote_mounts = {
+                mnt["Destination"]
+                for mnt in active_mounts
+            }
+            # Ignore ray bootstrap files.
+            requested_remote_mounts = {
+                self._docker_expand_user(remote)
+                for remote in cleaned_bind_mounts.keys()
+            }
+            unfulfilled_mounts = (
+                requested_remote_mounts - active_remote_mounts)
+            if unfulfilled_mounts:
+                try:
+                    re_init_required = cli_logger.confirm(
+                        False, "This Docker Container is already running, Do "
+                        "you want to restart the Docker container on "
+                        "this node? to pick up the following file_mounts {}",
+                        unfulfilled_mounts)
```

Comment:
Can we remove this confirm? This introduces interactivity/prompting, even when I do `ray up -y`.

Reply:
Yep! Merged in your change!